src/client/datascience/kernel-launcher/types.ts

```diff
@@ -110,9 +110,13 @@ export class PythonKernelDiedError extends BaseError {
     public readonly exitCode: number;
     public readonly reason?: string;
     constructor(options: { exitCode: number; reason?: string; stdErr: string } | { error: Error; stdErr: string }) {
+        // Last line in stack traces generally contains the error message.
+        // Display that in the error message.
+        let reason = ('reason' in options ? options.reason || '' : options.stdErr).trim().split('\n').reverse()[0];
+        reason = reason ? `${reason}, \n` : '';
         const message =
             'exitCode' in options
-                ? `Kernel died with exit code ${options.exitCode}. ${options.reason}`
+                ? `Kernel died (code: ${options.exitCode}). ${reason}${options.reason}`
```

Comment:
Updated to incldue error message in the first line (in python stack traces the error message is at the end, this way we extract the last line & put it on top of the error message)

Reply:
What does it look like in the UI?

Reply:
Sorry, meant to upload that, took the screenshots earlier.
here they are
![Screen Shot 2021-09-23 at 14 42 11](https://user-images.githubusercontent.com/1948812/134591934-5a185bfe-3106-4ceb-b375-07a771666bfb.png)
![Screen Shot 2021-09-23 at 14 50 40](https://user-images.githubusercontent.com/1948812/134591937-5ca677c5-f2b6-41ef-a726-eb7d7fceeb6b.png)